src/Temporalio/Bridge/include/temporal-sdk-bridge.h

```diff
@@ -353,6 +353,45 @@ typedef struct WorkerOrFail {
   const struct ByteArray *fail;
 } WorkerOrFail;
 
+typedef struct FixedSizeSlotSupplier {
+  uintptr_t num_slots;
+} FixedSizeSlotSupplier;
+
+typedef struct ResourceBasedTunerConfig {
+  double target_memory_usage;
+  double target_cpu_usage;
+} ResourceBasedTunerConfig;
+
+typedef struct ResourceBasedSlotSupplier {
+  uintptr_t minimum_slots;
+  uintptr_t maximum_slots;
+  uint64_t ramp_throttle_ms;
+  struct ResourceBasedTunerConfig tuner_config;
+} ResourceBasedSlotSupplier;
+
+typedef enum SlotSupplier_Tag {
+  FixedSize,
+  ResourceBased,
+} SlotSupplier_Tag;
+
+typedef struct SlotSupplier {
+  SlotSupplier_Tag tag;
+  union {
+    struct {
+      struct FixedSizeSlotSupplier fixed_size;
+    };
+    struct {
+      struct ResourceBasedSlotSupplier resource_based;
+    };
+  };
+} SlotSupplier;
```

Comment:
I think this makes our header file a bit uglier to consume from C compared to mutually exclusive options

Reply:
:man_shrugging: Eh, not really a problem IMO. If we care about the header being as easy as possible we can do that when we have another consumer, but honestly I think this is pretty easy to use as-is.